test/e2e/lb_eip_test.go

```diff
@@ -0,0 +1,572 @@
+	t.Helper()
+	t.Logf("effectuating eipAllocations for IngressController %s", ic.Name)
+	icName := types.NamespacedName{Name: ic.Name, Namespace: ic.Namespace}
+	progressingTrue := operatorv1.OperatorCondition{
+		Type:   ingress.IngressControllerLoadBalancerProgressingConditionType,
+		Status: operatorv1.ConditionTrue,
+	}
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, progressingTrue); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	// Delete and recreate the IngressController service to effectuate.
+	t.Logf("recreating the service to effectuate the subnets: %s/%s", controller.LoadBalancerServiceName(ic).Namespace, controller.LoadBalancerServiceName(ic).Namespace)
+	if err := recreateIngressControllerService(t, ic); err != nil {
+		t.Fatalf("failed to delete and recreate service: %v", err)
+	}
+
+	// Ensure the service's load-balancer status changes, and verify we get the expected eipAllocation annotation on the service.
+	waitForEIPAllocationAnnotation(t, ic, expectedEIPAllocations)
+
+	// Expect the load balancer to provision successfully with the new eipAllocations.
+	if err := waitForIngressControllerCondition(t, kclient, 10*time.Minute, icName, expectedOperatorConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	// Verify the subnets status field is configured to what we expect.
+	verifyIngressControllerEIPAllocationStatus(t, icName)
+}
+
+// createInvalidEIPAllocations fetches the public subnets and creates same number of invalid eipAllocations as that of public subnets.
+func createInvalidEIPAllocations(t *testing.T, ec2ServiceClient *ec2.EC2, vpcID string) ([]operatorv1.EIPAllocation, error) {
+	t.Helper()
+	publicSubnets, err := getPublicSubnets(vpcID, ec2ServiceClient)
+	if err != nil {
+		t.Fatalf("public subnets not created due to error: %v", err)
+	}
+	invalidEIPAllocations := make([]operatorv1.EIPAllocation, len(publicSubnets))
+	var hexString string
+	for i := 0; i < len(publicSubnets); i++ {
+		if hexString, err = randomHex(9); err != nil {
+			t.Fatalf("failed to create hexadecimal string for eipAllocation due to error: %v", err)
+		}
+		invalidEIPAllocations[i] = operatorv1.EIPAllocation(fmt.Sprintf("%s-%s", "eipalloc", hexString[:17]))
+	}
+	return invalidEIPAllocations, nil
+}
+
+func randomHex(n int) (string, error) {
+	bytes := make([]byte, n)
+	if _, err := random.Read(bytes); err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(bytes), nil
+}
+
+// cleanupEIPAllocations clean all unassociated EIPs created during the e2e tests. It returns true if no associated eipAllocations tagged with cluster name are found.
+func cleanupEIPAllocations(t *testing.T, svc *ec2.EC2, clusterName string, vpcID string) bool {
+	t.Log("Releasing unassociated EIPs")
+
+	tagKeyEIP := fmt.Sprintf("NE1674-%s", t.Name())
```

Comment:
This tag key is created in 2 places and has to be identical. Can you create a function which returns the same tag key for the creation and the cleanup of EIPs?